src/hotspot/share/opto/printinlining.cpp

```diff
@@ -46,7 +46,10 @@ void InlinePrinter::print_on(outputStream* tty) const {
   if (!is_enabled()) {
     return;
   }
-  _root.dump(tty, -1);
+  // using stringStream prevents interleaving with multiple compile threads
+  stringStream ss;
+  _root.dump(&ss, -1);
+  tty->print_raw(ss.freeze());
```

Comment:
General thought: I see that we use the proposed pattern to print a `stringStream` in existing code but also a different pattern with `as_string()`:
https://github.com/openjdk/jdk/blob/c56fb0b6eff7d3f36bc65f300b784e0dd73c563e/src/hotspot/share/opto/compile.cpp#L614

Can anybody comment on which one should be preferred?